File: src/main/java/com/zerozero/store/domain/service/StoreSearcher.java
package com.zerozero.store.domain.service;

import com.zerozero.store.domain.response.StoreSearchResponse;

import java.util.List;

public interface StoreSearcher {

    List<StoreSearchResponse> search(String query);

    List<StoreSearchResponse> searchByLocation(String query, double longitude, double latitude);
}

File: src/main/java/com/zerozero/store/domain/service/StoreSearcher.kt
package com.zerozero.store.domain.service

import com.zerozero.store.domain.response.StoreSearchResponse

interface StoreSearcher {
    fun search(query: String): List<StoreSearchResponse>

    fun searchByLocation(query: String, longitude: Double, latitude: Double): List<StoreSearchResponse>
}
